src/components/sheet/sheet-action-buttons/SendActionButton.js

```diff
@@ -1,24 +1,19 @@
 import lang from 'i18n-js';
 import React, { useCallback } from 'react';
 import SheetActionButton from './SheetActionButton';
-import { useExpandedStateNavigation } from '@/hooks';
 import Routes from '@/navigation/routesNames';
 import { IS_IOS } from '@/env';
+import useNavigationForNonReadOnlyWallets from '@/hooks/useNavigationForNonReadOnlyWallets';
 
 function SendActionButton({ asset, color: givenColor, ...props }) {
   const { colors } = useTheme();
   const color = givenColor || colors.paleBlue;
-  const navigate = useExpandedStateNavigation(null, false, asset);
+  const navigate = useNavigationForNonReadOnlyWallets();
   const handlePress = useCallback(
     () =>
-      navigate(Routes.SEND_FLOW, params => {
-        const updatedParams = { ...params, asset };
-        return IS_IOS
-          ? {
-              params: updatedParams,
-              screen: Routes.SEND_SHEET,
-            }
-          : { ...updatedParams };
+      navigate(Routes.SEND_FLOW, {
+        asset,
```

Comment:
does this mean we don't need this `asset` on line 15 anymore?

Reply:
we do for Android